lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3205,33 +3193,51 @@ static bool checkTransposingParameters(
 
 // SWIFT_ENABLE_TENSORFLOW
 void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
+  // Call `getParameterIndices` to trigger a
+  // `DifferentiableAttributeParameterIndicesRequest`, which currently performs
+  // full `@differentiable` type-checking.
+  // TODO: Consider creating separate requests for the following functionality:
+  // - `DifferentiableAttr::getJVPFunction`
+  // - `DifferentiableAttr::getVJPFunction`
+  // - `DifferentiableAttr::getDerivativeGenericSignature`
+  (void)attr->getParameterIndices();
+}
+
+llvm::Expected<IndexSubset *>
+DifferentiableAttributeParameterIndicesRequest::evaluate(
+    Evaluator &evaluator, DifferentiableAttr *attr, Decl *D) const {
   // Skip checking implicit `@differentiable` attributes. We currently assume
   // that all implicit `@differentiable` attributes are valid.
   // Motivation: some implicit attributes do not contain a where clause, and
   // this function assumes that the where clauses are available. Propagating
   // where clauses and requirements consistently is a larger problem, to be
   // revisited.
   if (attr->isImplicit())
```

Comment:
I think that with this change, this condition might no longer trigger, because the evaluator will see that the indices already exist. If so, it's a quite nice simplification! Could you switch this to an assertion that `!attr->isImplicit()`?